Protobuf changes to allow rotating creds

tcsc opened the pull request:
Adds protobuf description of a service for updating Plugins' static credentials.

The target credential can be addressed one of two ways:

1. directly by name, which is useful for updating it via IaC tools like terraform, and
2. by label matching, which is useful when you only know the name of the owning plugin resource. This will be the case for most human facing tools.

See-Also: #57691, gravitational/teleport.e#7016

smallinsky commented on api/proto/teleport/legacy/types/types.proto:
```diff
@@ -7981,6 +7996,8 @@ message PluginStaticCredentialsOAuthClientSecret {
 // PluginStaticCredentialsSSHCertAuthorities contains the active SSH CAs used
 // for the integration or plugin.
 message PluginStaticCredentialsSSHCertAuthorities {
+  option (gogoproto.equal) = true;
```
Instead of replying on the equal can we just consider compering the creds revision version ?

smallinsky replied:
> I also wouldn't like to rely on gogoproto features if that can be avoided. Maybe we can hand roll equality function?

I feel like this is overcomplicated. Sorry for confusion.  If a new creds will be added I'm sure that Equal function won't be extended. I have just asked if the revision can be used instead but right now it looks like we have Equal function that we need to maintenance.  I'm OK to use gogoproto.equal since this method is already used to comare plugins.

kopiczko reviewed (commented):
LGTM but let's address @smallinsky comment about UpdatePluginStaticCredentialsResponse

flyinghermit reviewed (commented):
Looks good but can we update PR description explaining how this whole credential update system will work? Or maybe a link to parent PR/ticket that explains it? 

I will only be reviewing names and typos otherwise :)

smallinsky reviewed (commented):
LGTM when a remains comment are resolved and PR description is added.

flyinghermit reviewed (approved):
lgtm when existing comments are resolved.

tcsc committed:
Adds protobuf description of a service for updating Plugins' static credentials.

The target credential can be addressed one of two ways:

1. directly by name, which is useful for updating it via IaC tools like terraform, and
2. by label matching, which is useful when you only know the name of the owning plugin
   resource. This will be the case for most human facing tools.

See-Also: #57691, gravitational/teleport.e#7016

tcsc committed:
Merge branch 'master' into tcsc/rotate-plugin-credentiels-protobuff
